src/LocalField/neq.jl

```diff
@@ -0,0 +1,216 @@
+
+degree_relative(L::FqNmodFiniteField, K::Hecke.Nemo.GaloisField)  = Int(absolute_degree(L)//absolute_degree(K))
+
+
+##############################################
+#random element with small coefficients
+##############################################
+
+function random_elem(L::Hecke.LocalField)
+     b = basis(L)
+     n = degree(L)
+     r = [rand(1:5*n) for i in 1:n]
+return sum( [ r[i]*b[i] for i in 1:n])
+end
+
+
+function random_elem(L::FlintQadicField)
+     b = basis_qadic(L)
+     n = degree(L)
+     r = [rand(1:5*n) for i in 1:n]
+return sum( [ r[i]*b[i] for i in 1:n])
+end
+
+
+
+
+
+function one_root( f::fq_nmod_poly, F:: Hecke.RelFinField)
+   g = polynomial(F, [coeff(f,i) for i = 0:degree(f) ] )
+   fac = factor(g)
+   if length(fac)  == 1
+      return []
+   end
+   r = [x for x in fac ][1][1]
+   @assert degree(r) == 1
+return -coeff(r,0)
+end
+
+
+function one_root( f::gfp_poly, F:: FqNmodFiniteField)
+   g = polynomial(F, [coeff(f,i) for i = 0:degree(f) ] )
+   fac = factor(g)
+   if length(fac)  == 1
+      return []
+   end
+   r = [x for x in fac ][1][1]
+   @assert degree(r) == 1
+return -coeff(r,0)
+end
+
+
+
+function one_root( f::Hecke.AbstractAlgebra.Generic.Poly, F:: Hecke.RelFinField)
+   g = polynomial(F, [coeff(f,i) for i = 0:degree(f) ] )
+   fac = factor(g)
+   if length(fac)  == 1
+      return []
+   end
+   r = [x for x in fac ][1][1]
+   @assert degree(r) == 1
+return -coeff(r,0)
+end
+
+function one_root( f::gfp_poly, F:: Hecke.RelFinField)
+   g = polynomial(F, [coeff(f,i) for i = 0:degree(f) ] )
+   fac = factor(g)
+   if length(fac)  == 1
+      return []
+   end
+   r = [x for x in fac ][1][1]
+   @assert degree(r) == 1
+return -coeff(r,0)
+end
+
+
+#########################################################################
+# Alternative way of computing the norm due to some problem in resultant
+#
+##########################################################################
+
+function norm_det(a)
+   L = parent(a)
+   m = matrix([[coeff(a*x, i-1) for i in 1:degree(L)] for x in basis(L)])
+return det(m)
+end
+
+
+######################### norm equation over finite fields ##############
+
+function norm_equation(F::FqNmodFiniteField, b::Union{gfp_elem, fq_nmod})
+    k = parent(b)
+    n = degree_relative(F,k)
+    Qx,x = PolynomialRing(k,"x")
+    Qz,z = PolynomialRing(F,"z")
+    f = polynomial(k,vcat([b],[rand(k) for i = 1:n-1],[1] ) )
+    while !isirreducible(f)
+       f = polynomial(k,vcat([b],[rand(k) for i = 1:n-1],[1] ) )
+    end
+return one_root(f,F)
+end
+
+function norm_equation(F::Hecke.RelFinField, b::gfp_elem)
+    k = parent(b)
+    n = degree_relative(F,k)
+    Qx,x = PolynomialRing(k,"x")
+    Qz,z = PolynomialRing(F,"z")
+    f = polynomial(k,vcat([b],[rand(k) for i = 1:n-1],[1] ) )
+    while !isirreducible(f)
+       f = polynomial(k,vcat([b],[rand(k) for i = 1:n-1],[1] ) )
+    end
+return one_root(f,F)
+end
+
+
+
+function norm_equation(F::Hecke.RelFinField, b::fq_nmod)
```

Comment:
Again: aren't you duplicating the same code, just with different types for the arguments? If so, once again: merge them and use a union type for the arguments?